build_tools/cmake/iree_benchmark_suite.cmake

```diff
@@ -237,10 +237,51 @@ function(iree_benchmark_suite)
         add_dependencies(iree-benchmark-suites "${_TRANSLATION_TARGET_NAME}")
       endif(NOT TARGET "${_TRANSLATION_TARGET_NAME}")
 
+      set(_COMPILATION_STATS_TRANSLATION_TARGET_NAME
+        "${_TRANSLATION_TARGET_NAME}-compilation-stats"
+      )
+      set(_COMPILATION_STATS_VMFB_FILE
+        "${_VMFB_ARTIFACTS_DIR}/${_MODULE_SOURCE_BASENAME_WITH_HASH}-compilation-stats.vmfb"
+      )
+      if(IREE_ENABLE_COMPILATION_BENCHMARKS AND NOT TARGET "${_COMPILATION_STATS_TRANSLATION_TARGET_NAME}")
+        iree_bytecode_module(
+          NAME
+            "${_COMPILATION_STATS_TRANSLATION_TARGET_NAME}"
+          MODULE_FILE_NAME
+            "${_COMPILATION_STATS_VMFB_FILE}"
+          SRC
+            "${_MODULE_SOURCE}"
+          FLAGS
+            "--iree-vm-emit-polyglot-zip=true"
+            "--iree-llvm-debug-symbols=false"
+            ${_TRANSLATION_ARGS}
+          DEPENDS
+            "${_MODULE_SOURCE_TARGET}"
+          FRIENDLY_NAME
+            "${_FRIENDLY_TARGET_NAME}"
+        )
+
+        # TODO: that add_custom_target should be done by iree_bytecode_module.
+        # Note: at the moment, iree_bytecode_module doesn't do much with its
+        # NAME argument.
+        add_custom_target("${_COMPILATION_STATS_TRANSLATION_TARGET_NAME}"
+          DEPENDS "${_COMPILATION_STATS_VMFB_FILE}"
+        )
```

Comment:
I'm not quite following this comment

`NAME` is used here:
https://github.com/google/iree/blob/794bbf4c23e790d9f897d7b6eebd0522e7c4af47/build_tools/cmake/iree_bytecode_module.cmake#L123-L126

We could add a target for the .vmfb generation (as well as the .c generation), as you have it here. That would be most flexible, but would add a bunch of new, partially redundant, CMake targets. As there are real uses for those targets, I think I'd be fine with just adding them, rather than putting special case logic here in `iree_benchmark_suite.cmake`.

Reply:
This is also an issue for building the current benchmark modules. I think it's better to add them in the `iree_bytecode_module`, but that requires some extra changes and touches unrelated code.

I filed #9254 for this and updated the comment here.